packages/create-hydrogen/tsup.config.ts

```diff
@@ -1,13 +1,38 @@
+import {createRequire} from 'node:module';
+import fs from 'node:fs/promises';
 import {defineConfig} from 'tsup';
 
 export default defineConfig({
-  entry: ['src/**/*.ts'],
+  entry: ['src/create-app.ts'],
   outDir: 'dist',
   format: 'esm',
-  minify: false,
-  bundle: false,
-  splitting: true,
-  treeshake: true,
+  clean: true,
   sourcemap: false,
   dts: false,
+  minify: true,
+  splitting: true, // Async/await breaks without splitting
+
+  // -- Bundle:
+  bundle: true,
+  external: [
+    '@ast-grep/napi', // Required binary
+    '@parcel/watcher', // Not used but but needs to resolve it
+    'react-devtools-core', // Not used but breaks the build otherwise
+  ],
+  noExternal: ['@shopify/cli-hydrogen'],
```

Comment:
do you need to specify that this is noExternal? isn't everything "no external" by default?